File: game_server/ClientProtocol/GC2GS/p004_PlayerOp/GC2GS_004_013_ReqReportPlayer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GC2GS.p004_PlayerOp
{

/// <summary>
/// 举报指定玩家
/// </summary>
public class GC2GS_004_013_ReqReportPlayer : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 举报目标玩家CID
/// </summary>
private long targetCid;
/// <summary>
/// 举报内容
/// </summary>
private string content;


public GC2GS_004_013_ReqReportPlayer() {
	targetCid = (long)0;
	content = "";
}

public GC2GS_004_013_ReqReportPlayer(
	long _targetCid
	, string _content
) {	targetCid = _targetCid;
	content = _content;
}

public byte getMainOrder() { return (byte)4; }

public byte getSubOrder() { return (byte)13; }

/// <summary>
/// 举报目标玩家CID
/// </summary>
public long getTargetCid() { return targetCid; }
/// <summary>
/// 举报目标玩家CID
/// </summary>
public void setTargetCid(long _targetCid) { targetCid = _targetCid; }
/// <summary>
/// 举报内容
/// </summary>
public string getContent() { return content; }
/// <summary>
/// 举报内容
/// </summary>
public void setContent(string _content) { content = _content; }


public int GetBufSize() {
	int _size = 8;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(content);

	return _size;
}

public int GetFullPackBufSize() {
	int _size = 10;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(content);

	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	targetCid = _buf.getLong();
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	content = _buf.getString();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.putLong(targetCid);
	_buf.putString(content);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)13);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)13);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("targetCid").Append(":").Append(targetCid.ToString()).Append(", ");
	builder.Append("content").Append(":").Append(content.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GC2GS/p004_PlayerOp/GC2GS_004_013_ReqReportPlayer.java
package GC2GS.p004_PlayerOp;

import java.nio.ByteBuffer;
/*********
 * 举报指定玩家
 **/
public class GC2GS_004_013_ReqReportPlayer implements ALBasicProtocolPack._IALProtocolStructure {
/** 举报目标玩家CID */
private long targetCid;
/** 举报内容 */
private String content;


public GC2GS_004_013_ReqReportPlayer() {
	targetCid = (long)0;
	content = "";
}

public GC2GS_004_013_ReqReportPlayer(
	 long _targetCid
	, String _content
) {	targetCid = _targetCid;
	content = _content;
}

public final byte getMainOrder() { return (byte)4; }

public final byte getSubOrder() { return (byte)13; }

/** 举报目标玩家CID */
public long getTargetCid() { return targetCid; }
/** 举报目标玩家CID */
public void setTargetCid(long _targetCid) { targetCid = _targetCid; }
/** 举报内容 */
public String getContent() { return content; }
/** 举报内容 */
public void setContent(String _content) { content = _content; }


public final int GetBufSize() {
	int _size = 8;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(content);

	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 10;
	_size += ALBasicProtocolPack.ALProtocolCommon.GetStringBufSize(content);

	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) targetCid = _buf.getLong();
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) content = ALBasicProtocolPack.ALProtocolCommon.GetStringFromBuf(_buf);
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.putLong(targetCid);
	ALBasicProtocolPack.ALProtocolCommon.PutStringIntoBuf(_buf, content);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)4);
	_buf.put((byte)13);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)4);
	_recBuf.put((byte)13);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

